typescript/cli/src/config/ism.ts

```diff
@@ -106,68 +111,51 @@ export async function createAdvancedIsmConfig(
     case IsmType.FALLBACK_ROUTING:
       return createFallbackRoutingConfig(context);
     case IsmType.MERKLE_ROOT_MULTISIG:
-      return createMerkleRootMultisigConfig(context);
     case IsmType.MESSAGE_ID_MULTISIG:
-      return createMessageIdMultisigConfig(context);
+    case IsmType.STORAGE_MERKLE_ROOT_MULTISIG:
+    case IsmType.STORAGE_MESSAGE_ID_MULTISIG:
+      return createMultisigConfig(moduleType);
     case IsmType.ROUTING:
       return createRoutingConfig(context);
     case IsmType.TEST_ISM:
       return { type: IsmType.TEST_ISM };
     case IsmType.TRUSTED_RELAYER:
       return createTrustedRelayerConfig(context, true);
     default:
-      throw new Error(`Invalid ISM type: ${moduleType}.`);
+      throw new Error(`Unsupported ISM type: ${moduleType}.`);
   }
 }
 
-export const createMerkleRootMultisigConfig = callWithConfigCreationLogs(
-  async (): Promise<MultisigIsmConfig> => {
-    const validatorsInput = await input({
-      message:
-        'Enter validator addresses (comma separated list) for merkle root multisig ISM:',
-    });
-    const validators = validatorsInput.split(',').map((v) => v.trim());
-    const thresholdInput = await input({
-      message:
-        'Enter threshold of validators (number) for merkle root multisig ISM:',
-    });
-    const threshold = parseInt(thresholdInput, 10);
-    if (threshold > validators.length) {
-      errorRed(
-        `Merkle root multisig signer threshold (${threshold}) cannot be greater than total number of validators (${validators.length}).`,
-      );
-      throw new Error('Invalid protocol fee.');
-    }
-    return {
-      type: IsmType.MERKLE_ROOT_MULTISIG,
-      threshold,
-      validators,
-    };
-  },
-  IsmType.MERKLE_ROOT_MULTISIG,
-);
-
-export const createMessageIdMultisigConfig = callWithConfigCreationLogs(
-  async (): Promise<MultisigIsmConfig> => {
-    const thresholdInput = await input({
-      message:
-        'Enter threshold of validators (number) for message ID multisig ISM',
-    });
-    const threshold = parseInt(thresholdInput, 10);
+export const createMultisigConfig = async (
+  ismType: MultisigIsmConfig['type'],
+): Promise<MultisigIsmConfig> => {
+  const validatorsInput = await input({
+    message:
+      'Enter validator addresses (comma separated list) for multisig ISM:',
+  });
+  const validators = validatorsInput.split(',').map((v) => v.trim());
+  const threshold = parseInt(
```

Comment:
maybe also check the given threshold >=1 and <= validators.length

should we also recommend a value?

Reply:
trying to leave input validation to the zod schema
can make this change there I suppose though 0/n as a noop ISM could be useful